File: regula_falsi.py
# regula_falsi.py

def f(x):
    return x**3 - x - 2

a = float(input("Enter a: "))
b = float(input("Enter b: "))

if f(a) * f(b) >= 0:
    print("Invalid interval")
    exit()

i = 0

while True:
    c = (a*f(b) - b*f(a)) / (f(b) - f(a))

    if abs(f(c)) < 0.0001:
        break

    if f(a) * f(c) < 0:
        b = c
    else:
        a = c

    i += 1

print("Root:", c)
print("Iterations:", i)

File: regula_falsi.c
#include <stdio.h>
#include <math.h>

float f(float x) {
    return x*x*x - x - 2;   // example function
}

int main() {
    float a, b, c;
    int i = 0;

    printf("Enter a and b: ");
    scanf("%f %f", &a, &b);

    if(f(a) * f(b) >= 0) {
        printf("Invalid interval\n");
        return 0;
    }

    do {
        c = (a * f(b) - b * f(a)) / (f(b) - f(a));

        if(f(c) == 0.0)
            break;
        else if(f(a) * f(c) < 0)
            b = c;
        else
            a = c;

        i++;

    } while(fabs(f(c)) >= 0.0001);

    printf("Root = %.4f\n", c);
    printf("Iterations = %d\n", i);

    return 0;
}
